### PR TITLE
[FIX] web: add preparse method for parsing dates in catalan

### PR DESCRIPTION
**Pull request by anhe-odoo**
### Observed Behaviour
When opening a pivot view with the catalan language activated and grouping rows/cols by months, everything works fine. But if
we try to import this pivot view in a spreadsheet, all months are converted to gener, which means 'January' in catalan

### Expected Behaviour
When importing the pivot view in a spreadsheet, we should get the correct date grouping

### Reproducibility
This issue can be reproduced with the following steps
1. Activate the catalan language and set it
2. Got to Sales App
3. Select the pivot view
4. Group the rows/cols by order date > months
5. Try to import the pivot view in a spreadsheet

### Fix Description
The isse was coming from the fact babel (the package used to generate date as text in python) write the month as 'de gener',
'de febrer', 'de març', 'd'abril', while moment (the lib used to handle date in JS) is looking for months as 'gener', 'febrer',
... (ie without "de" or "d'" prefix), causing it not to recognize the date proposed by Babel. We then added a preparse
method to the catalan language, automatically used by moment when present in the language definition.

### Related Issues/PR
 - opw-2817041

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by anhe-odoo** [comment]
Hello @odoo/rd-framework-js : The customer came back to me as some of his clients are experiencing the addressed issue. May I ask you to have a look at this one please ? Thanks :)

**Comment by kebeclibre** [comment]
We usually don't want to modify a lib in place, because if we want to update the lib's code (same version bugfixes) it'd conflict.

It is still possible to make your fix in odoo's code with moment's API (I think it is something like moment.locale(....), not sure though).

Maybe the same issue/bugfix exists on moment's repo, did you check that ?

finally it seems weird to have a special case that shed a light on some cumbersome coupling between client and server

**Comment by anhe-odoo** [comment]
Yes, the issue is existing in moment repo (see here : https://github.com/moment/moment/issues/4508), but it has been closed without any solution. I'll have a look at this moment.locale thing, thanks :)

File: addons/web/static/src/js/libs/moment.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+'use strict';
+const locale = moment.locale();
+moment.updateLocale('ca', {
+    preparse: function (string) {
+        return string.replace(/\b(?:d’|de )(gener|febrer|març|abril|maig|juny|juliol|agost|setembre|octubre|novembre|desembre)/g, '$1');
+    }
+});
+if(locale !== 'ca'){
+    moment.locale(locale);
+}
```
Comment (yelizariev) [inline addons/web/static/src/js/libs/moment.js:17]: What is purpose of this code?

Comment (anhe-odoo) [reply addons/web/static/src/js/libs/moment.js:17]: First we update the catalan localization to replace "de gener", "de febrer", ... to "gener", "febrer", ...
Then, as `moment.updateLocale('ca', ...)` will set the current localization to Catalan, we reset it to the previous local (but only if the previous wasn't the catalan language)

Comment (yelizariev) [inline addons/web/static/src/js/libs/moment.js:17]: Maybe we could update Catalan localization only when it's used?
```suggestion
if(locale === 'ca'){
    moment.updateLocale('ca', {
        preparse: function (string) {
            return string.replace(/\b(?:d’|de )(gener|febrer|març|abril|maig|juny|juliol|agost|setembre|octubre|novembre|desembre)/g, '$1');
        }
    });
}
```

Comment (anhe-odoo) [reply addons/web/static/src/js/libs/moment.js:17]: Indeed, this was my first try, but (if m'y understanding is correct) this part of the code is only executed when starting the server, meaning that if someone start with the English language and then change its language to Catalan, this won't work until he restarts the server.

Comment (yelizariev) [reply addons/web/static/src/js/libs/moment.js:17]: Why would you need to restart the server? You probably mean refresh web page? But it's already mandatory action on switching to another language

**Review by kebeclibre — APPROVED** [review]

**Review by nle-odoo — APPROVED** [review]
robodoo r+

**Comment by nle-odoo** [comment]
> staging failed: ci/runbot on 9c386655e797910b1c06c25ed0c4276df576c6ab (view more at https://runbot.odoo.com/runbot/build/19067917)
this is an entreprise hr payrol error that doesn't seem related to catalan language

robodoo retry

**Pull request closed without merging**